posts/2021/05/rethinking-jupyter-documentation.md

```diff
@@ -0,0 +1,192 @@
+browser context (notebook, JupyterLab, Spyder) with proper rendering, and currently
+understand most directives; it could be customized to understand any new ones:
+
+<img alt="papyri1" src="/images/2021/05/papyri-1.png" class='center' >
+
+Above is the (terminal) documentation of `scipy.polynomial.lagfit`, see how the
+single backticks are properly understood and refer to known parameters, it
+detected that  `` `n` `` is incorrect as it should have double backticks; notice
+the rendering of the math even in terminal.
+
+For that matter technically this does not care as to whether the DocString is
+written in RST or Markdown; though I need to implement the later part. I believe
+though that some maintainers would be quite happy to use markdown whose syntax
+more users are familiar with.
+
+<img alt="papyri navigation" src="/images/2021/05/papyri-nav.gif" class='center' >
+
+It support navigation – here in terminal – where clicking or pressing enter on a
+link would bring you to the target page. In above gif you can see that many
+token of code example are also automatically type-inferred (thanks [Jedi](https://github.com/davidhalter/jedi)), and
+can also be clicked to navigate to their corresponding page.
+
+<img alt="papyri terminal-fig" src="/images/2021/05/papyri-terminal-fig.png" class='center' >
+
+Images are included, even in terminal when they are not inline but replaced by
+a button to open them in your preferred viewer (see the `Open with quicklook` in
+above screenshot).
+
+# The future
+
+
+I'm working on a number of other features, in particular :
+
+ - rendering of narrative docs – for which I have a prototype,
+ - automatic indexing of all the figures and plots –  working but slow right now.
+ - proper cross library reference and indexing without the need for intersphinx.
+   for example It is possible from the `numpy.linspace` page to see all page that
+   reference it, or use `numpy.linspace` in their example section
+   (see previous image).
+
+And many others, like showing a graph of the local references between functions,
+search, and preference configurability. I think this could also support many
+other desirable features, like user preferences (hide/show type annotation,
+deprecated directives, and custom coloration/syntax), though haven't started
+working on these, and I have some ideas on how this could be uses to provide
+translations as well.
+
+Right now is it not as fast as efficient as I would like to – though it's faster
+than running Sphinx on the fly – but required some ahead of time processing. And
+crash in many places; It can render most of the documentation of scipy, numpy,
+xarray, IPython and scikit image.
+
+I though encourage you to think about what features you are missing when using
+documentation from withing Jupyter and let me know. I hope this could becomme a
+nice addition to sphinx when consulting documentation from within Jupyter.
```

Comment:
I think there is a stray word here, but I may have misunderstood:

```suggestion
I encourage you to think about what features you are missing when using
documentation from within Jupyter and let me know. I hope this could become a
nice addition to Sphinx when consulting documentation from within Jupyter.
```